Added EmptyList component to show message and Add button

h-kataria opened the pull request:
This component will be displayed when there are no records to display on the page, along with Add button(if user has access to Add new records)
https://issues.redhat.com/browse/AA-543
![Screen Shot 2021-06-10 at 11 23 53 AM](https://user-images.githubusercontent.com/3450808/121552389-6dfaae80-c9de-11eb-8885-d6989278ccc2.png)

brumik commented on src/Components/EmptyList.js:
```diff
@@ -0,0 +1,41 @@
+  )
+};
+
+export default EmptyList;
```
[nitpick] Missing newline

brumik commented on src/Components/EmptyList.test.js:
```diff
@@ -0,0 +1,32 @@
+    expect(screen.getByRole('button')).toBeTruthy;
+    expect(screen.getByText('Add plan')).toBeTruthy();
+  });
+});
```
[nitpick] Missing new line

brumik commented on src/Components/EmptyList.js:
```diff
@@ -0,0 +1,41 @@
+import { AddCircleOIcon } from '@patternfly/react-icons';
+import {useHistory, useLocation} from "react-router-dom";
+
+const EmptyList = ({ label = '', title = '', message = '', canAdd = 'false' }) => {
```
[nitpick] New lines (or use eslint fix command)

Also can add should be boolean instead of a string.

brumik commented on src/Components/EmptyList.js:
```diff
@@ -0,0 +1,41 @@
+    </EmptyState>
+  )
+};
+
```
Missing prop validation

brumik commented on src/Components/EmptyList.js:
```diff
@@ -0,0 +1,41 @@
+
+const EmptyList = ({ label = '', title = '', message = '', canAdd = 'false' }) => {
+  const history = useHistory();
+  const {pathname} = useLocation();
```
eslint fix ==> `{ pathname } = useLocation()`

brumik commented on src/Components/EmptyList.js:
```diff
@@ -0,0 +1,41 @@
+    <EmptyState variant="full">
+      <EmptyStateIcon icon={AddCircleOIcon}/>
+      <Title size="lg" headingLevel="h3">
+        {title || 'No items found.'}
```
Move this logic to the component title default value (eg.: Line 12 `... title = 'No items found' ...`)

brumik commented on src/Components/EmptyList.js:
```diff
@@ -0,0 +1,41 @@
+            aria-label={label}
+            onClick={() => {
+              history.push({
+                pathname: `${pathname}/add`,
```
Since it is located in the `Components` folder I think we want to reuse it, so I would probably pass the on-click handler or at least the pathname as a param to the component, so it is really general.

Also we could define the paths somewhere because composing the pathname from a variable and constant this deep is not too readable. Maybe the react router has something to offer.

brumik reviewed (changes requested):
It is nice a nice job, there are some linting errors (they do not show up as errors when running lint but locally the prettify wants to make some changes). The only real blocker is the on-click method, look in the comments :) 

Overall I like how it works :)

If you have some extra time, this is an easy component to make it work in typescript (you only need to add prop types to the component itself, so it is a good starter if you want to start to work with TS :)

brumik commented:
@h-kataria I was also able to compile it, but how you make no data so you can see it working? I was not able to do it.

h-kataria commented:
> @h-kataria I was also able to compile it, but how you make no data so you can see it working? I was not able to do it.

change https://github.com/RedHatInsights/tower-analytics-frontend/blob/AA-543-savings-planner-empty-view/src/Api.js#L29 to `return []`

brumik commented:
Verified, the component seems to work as it should. Thanks for the help ^^

h-kataria commented:
> Verified, the component seems to work as it should. Thanks for the help ^^

np, addressing other comments now

h-kataria commented:
> Verified, the component seems to work as it should. Thanks for the help ^^

@brumik thanks for testing/reviewing, addressed all the comment please re-review

brumik reviewed (approved):
Good to go :clap:
